### DISCUSSION TITLE
Public for subdomain

### DISCUSSION THREAD

**Discussion by sheddev** [community/community#22056] [Other Feature Feedback, Questions, & Ideas]
<p>This is probably a really ignorant question, but I have to ask it. I was looking at pointing my repository to a subdomain on my hosting server. From what I see you have to make it public. If that is true then isn’t your code exposed to anyone? If there is a flaw in your code it seems anyone could exploit it easily.</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Are you pointing a subdomain you own to a GitHub Pages site hosted from a private repository? If so, then yes, this is expected. Any files you have in publishing source of your site—either the <code>master</code> branch, the <code>gh-pages</code> branch, or the <code>docs</code> directory, depending on how your site is set up—will be publicly available on the internet. In order for your site to load a browser would need to be able to access and download the code for your site before rendering it on whatever device you were viewing the webpage from.</p>
<p>If you don’t want certain code exposed on the internet then you’ll need to move it out of the publishing source of your site—out of the <code>master</code> or <code>gh-pages</code> branches, depending on your chosen publishing source—or delete it entirely from your repository.</p>

**Comment by sheddev** [comment]
<p>Thanks for helping me to understand this.</p>
